src/main/java/com/fasterxml/jackson/core/io/JsonStringEncoder.java

```diff
@@ -137,7 +148,12 @@ public char[] quoteAsString(String input)
             return Arrays.copyOfRange(outputBuffer, 0, outPtr);
         }
         textBuffer.setCurrentLength(outPtr);
```

Comment:
I think we can replace combination with 

   return textBuffer.setCurrentAndReturn();